clients/src/test/java/org/apache/kafka/common/metrics/stats/FrequenciesTest.java

```diff
@@ -73,14 +72,14 @@ public void testMoreFrequencyParametersThanBuckets() {
     }
 
     @Test
-    public void testBooleanFrequencies() {
+    public void testBooleanFrequenciesStrategy1() {
```

Comment:
Just curious. It seems that we just break the existing tests into smaller ones. Did we change any logic for the tests to pass?

Reply:
There were 3 scenarios inside single test, which effectively should have been reset the state of metric in between. But reset was based on the knowledge of internal implementation of samples/windows, which is not a part of class API and which was slightly changed. Now it is separated to different test methods to make reset explicit.